Modsenfy.DataAccessLayer/Repositories/UserRepository.cs

```diff
@@ -107,5 +108,16 @@ public async Task<User> GetUserWithPlaylists(int id)
 
         return user;
     }
-    
+
+    public async Task<User> GetByUsername(string username)
+    {
+        return await _databaseContext.Users.FirstOrDefaultAsync(x => x.UserNickname ==  username);
+    }
+
+    public async Task<string> GetUserRole(User user)
+    {
+        var role = await _databaseContext.Role.FindAsync(user.UserRoleId);
+        if (role == null) { throw new NullReferenceException(); }
```

Comment:
Why do you throw an exception there? Are you considering to use **try catch** every time you call this method?